finance-service-api/src/test/kotlin/com/ajaxproject/financeservice/NatsControllerTest.kt

```diff
@@ -0,0 +1,188 @@
+package com.ajaxproject.financeservice
+
+import com.ajaxproject.financeservice.repository.FinanceRepositoryImpl
+import com.ajaxproject.internalapi.NatsSubject
+import com.ajaxproject.financeservice.dto.toProtoEnumFinance
+import com.ajaxproject.financeservice.dto.toProtoFinance
+import com.ajaxproject.financeservice.enums.Finance.EXPENSE
+import com.ajaxproject.financeservice.enums.Finance.INCOME
+import com.ajaxproject.financeservice.models.MongoFinance
+import com.ajaxproject.internalapi.finance.input.reqreply.CreateFinanceRequest
+import com.ajaxproject.internalapi.finance.input.reqreply.CreateFinanceResponse
+import com.ajaxproject.internalapi.finance.input.reqreply.DeleteFinanceByIdRequest
+import com.ajaxproject.internalapi.finance.input.reqreply.DeleteFinanceByIdResponse
+import com.ajaxproject.internalapi.finance.input.reqreply.GetAllFinancesByIdRequest
+import com.ajaxproject.internalapi.finance.input.reqreply.GetAllFinancesByIdResponse
+import com.ajaxproject.internalapi.finance.input.reqreply.GetCurrentBalanceRequest
+import com.ajaxproject.internalapi.finance.input.reqreply.GetCurrentBalanceResponse
+import com.google.protobuf.GeneratedMessageV3
+import com.google.protobuf.Parser
+import io.nats.client.Connection
+import org.assertj.core.api.Assertions.assertThat
+import org.bson.types.ObjectId
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.Test
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.data.mongodb.core.MongoTemplate
+import org.springframework.data.mongodb.core.query.Criteria
+import org.springframework.data.mongodb.core.query.Query
+import java.time.Duration
+import java.util.*
+
+
+@SpringBootTest
+class NatsControllerTest {
+
+    @Autowired
+    private lateinit var natsConnection: Connection
+
+    @Autowired
+    private lateinit var mongoTemplate: MongoTemplate
+
+    @Autowired
+    private lateinit var financeRepository: FinanceRepositoryImpl
+
+    @AfterEach
+    fun clean() {
+        val query = Query.query(Criteria.where("userId").`is`(4757839801))
+        mongoTemplate.remove(query, MongoFinance::class.java)
```

Comment:
don't you think `mongoTemplate.remove<MongoFinance>(query)` looks better?

Reply:
I agree

Reply:
It's not working